src/docpage.js

```diff
@@ -3,19 +3,201 @@ import React from 'react'
+        doc = state.doc || {},
+        loaded = state.loaded
+    let data = doc.content,
+        files = doc.children || []
+
+    if(!loaded) return <DocLoading />
+
+    return (
+      <div className="card doc-page">
+
+        <div className="lead">#{doc.id}: {data.filename}</div>
+
+        <div className="bg-faded doc-section-title">Meta</div>
+
+        <table className="table table-sm">
+          <tbody>
+            <tr>
+              <td style={{width: 110}}>Path</td>
+              <td><code>{data.path}</code></td>
+            </tr>
+            <tr>
+              <td>Filename</td>
+              <td><code>{data.filename}</code></td>
+            </tr>
+            {data.type &&
+              <tr>
+                <td>Type</td>
+                <td><code>{data.type}</code></td>
+              </tr>
+            }
+            {data.type != 'folder' && data.md5 &&
+              <tr>
+                <td>MD5</td>
+                <td><code>{data.md5}</code></td>
+              </tr>
+            }
+            {data.type != 'folder' && data.sha1 &&
+              <tr>
+                <td>SHA1</td>
+                <td><code>{data.sha1}</code></td>
+              </tr>
+            }
+            {data.lang &&
+              <tr>
+                <td>Language</td>
+                <td><code>{data.lang}</code></td>
+              </tr>
+            }
+            {data['date-created'] &&
+              <tr>
+                <td>Created</td>
+                <td>{data['date-created']}</td>
+              </tr>
+            }
+            {data.date &&
+              <tr>
+                <td>Modified</td>
+                <td>{data.date}</td>
+              </tr>
+            }
+            {data.pgp &&
+              <tr>
+                <td>PGP</td>
+                <td>{data.pgp}</td>
+              </tr>
+            }
+          </tbody>
+        </table>
+
+        <DocEmailSection doc={doc} />
+        <DocFilesSection title="Files" data={files} baseUrl={this.baseUrl} />
+        <DocTextSection data={doc.content.text} title="Text" />
+        <DocTextSection data={doc.content.tree} title="Headers &amp Parts"/>
+      </div>
+    )
+
+  }
+
+}
+
+class DocEmailSection extends React.Component {
+
+  render() {
+    let doc = this.props.doc || {},
+        data = doc.content,
+        files = doc.children || []
+
+    return data.type == 'email' && (
+      <div>
+        <p className="bg-faded doc-section-title">Email</p>
+        <table className="table table-sm">
+          <tbody>
+            <tr><td>From</td> <td>{data.from}</td></tr>
+            <tr><td>To</td> <td>{data.to.join(', ')}</td></tr>
+            <tr><td>Date</td> <td>{data.date}</td></tr>
+            <tr><td>Subject</td> <td>{data.subject || '---'}</td></tr>
+          </tbody>
+        </table>
+      </div>
+    )
+  }
+
+}
+
+class DocFilesSection extends React.Component {
+
+  render() {
+    let files = this.props.data.map((item, index) => {
```

Comment:
And there's another neat trick you can do - unpack `item` in the function arguments:

```javascript
data.map(({id, filename, content_type, size}, index) => ( <p>the value of item.id is {id}</p> ))
```